graylog2-server/src/main/java/org/graylog2/grok/GrokPattern.java

```diff
@@ -17,66 +17,78 @@
 package org.graylog2.grok;
 
 import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.auto.value.AutoValue;
 import com.google.common.base.MoreObjects;
-import org.bson.types.ObjectId;
 import org.mongojack.Id;
+import org.mongojack.ObjectId;
 
-import java.util.Objects;
+import javax.annotation.Nullable;
 
+@AutoValue
 @JsonAutoDetect
-public final class GrokPattern {
+public abstract class GrokPattern {
+
+    @JsonProperty("id")
+    @Nullable
     @Id
-    @org.mongojack.ObjectId
-    public ObjectId id;
-    public String name;
-    public String pattern;
-    public String contentPack;
-
-    public String name() {
-        return name;
-    }
+    @ObjectId
+    public abstract String id();
 
-    public String pattern() {
-        return pattern;
-    }
+    @JsonProperty
+    public abstract String name();
 
-    public String contentPack() {
-        return contentPack;
-    }
+    @JsonProperty
+    public abstract String pattern();
+
+    @JsonProperty
+    @Nullable
+    public abstract String contentPack();
 
     @Override
     public String toString() {
```

Comment:
AutoValue already produces a similar (if not identical) `toString()` method, so this one could be removed.